conn.go

```diff
@@ -340,9 +343,35 @@ func (s *Swarm) removeStream(stream *Stream, reset bool) error {
 	return err
 }
 
+// NOTE: must be called under the connIdxLock lock.
+func (s *Swarm) removeConnGroup(c *Conn, g Group) {
+	m, ok := s.connIdx[g]
+	if !ok {
+		return
+	}
+	delete(m, c)
+	if len(m) == 0 {
+		delete(c.swarm.connIdx, g)
+	}
+}
+
+// NOTE: must be called under the connIdxLock lock.
+func (s *Swarm) addConnGroup(c *Conn, g Group) {
```

Comment:
Does this need to be a method on the Swarm anymore? We seem to be accessing the swarm instance through the `Conn` now anyways